src/MOI_wrapper.jl

```diff
@@ -0,0 +1,355 @@
+    if !isnothing(model.manifold)
+        throw(
+            AddConstraintNotAllowed{F,typeof(set)}(
+                "Only one manifold allowed, variables in `$(model.manifold)` have already been added.",
+            ),
+        )
+    end
+    model.manifold = set.manifold
+    model.problem = nothing
+    model.state = nothing
+    n = MOI.dimension(set)
+    v = MOI.VariableIndex.(1:n)
+    for _ in 1:n
+        push!(model.variable_primal_start, nothing)
+    end
+    return v, MOI.ConstraintIndex{F,typeof(set)}(1)
+end
+
+function MOI.is_valid(model::Optimizer, vi::MOI.VariableIndex)
+    return !isnothing(model.manifold) &&
+           1 <= vi.value <= MOI.dimension(VectorizedManifold(model.manifold))
+end
+
+function MOI.get(model::Optimizer, ::MOI.NumberOfVariables)
+    if isnothing(model.manifold)
+        return 0
+    else
+        return MOI.dimension(VectorizedManifold(model.manifold))
+    end
+end
+
+function MOI.supports(::Optimizer, ::MOI.VariablePrimalStart, ::Type{MOI.VariableIndex})
+    return true
+end
+
+function MOI.set(
+    model::Optimizer,
+    ::MOI.VariablePrimalStart,
+    vi::MOI.VariableIndex,
+    value::Union{Real,Nothing},
+)
+    MOI.throw_if_not_valid(model, vi)
+    model.variable_primal_start[vi.value] = value
+    model.state = nothing
+    return nothing
+end
+
+MOI.supports(::Optimizer, ::MOI.NLPBlock) = true
+
+function MOI.set(model::Optimizer, ::MOI.NLPBlock, nlp_data::MOI.NLPBlockData)
+    model.nlp_data = nlp_data
+    model.problem = nothing
+    model.state = nothing
+    return nothing
+end
+
+function MOI.supports(
+    ::Optimizer, ::Union{MOI.ObjectiveSense,MOI.ObjectiveFunction{F}}
+) where {F<:_FUNCTIONS}
+    return true
+end
+
+function MOI.set(model::Optimizer, ::MOI.ObjectiveSense, sense::MOI.OptimizationSense)
+    model.sense = sense
+    return nothing
+end
+
+MOI.get(model::Optimizer, ::MOI.ObjectiveSense) = model.sense
+
+function MOI.get(
+    model::Optimizer, attr::Union{MOI.ObjectiveFunctionType,MOI.ObjectiveFunction}
+)
+    return MOI.get(model.qp_data, attr)
+end
+
+function MOI.set(
+    model::Optimizer, attr::MOI.ObjectiveFunction{F}, func::F
+) where {F<:_FUNCTIONS}
+    MOI.set(model.qp_data, attr, func)
+    model.problem = nothing
+    model.state = nothing
+    return nothing
+end
+
+"""
+    MOI.eval_objective(model::Manopt.Optimizer, x)
+
+Return the objective value at the vector of values `x` for the vectorization of
+the problem.
+"""
+function MOI.eval_objective(model::Optimizer, x)
+    if model.sense == MOI.FEASIBILITY_SENSE
+        return 0.0
+    elseif model.nlp_data.has_objective
+        return MOI.eval_objective(model.nlp_data.evaluator, x)
+    end
+    return MOI.eval_objective(model.qp_data, x)
+end
+
+"""
+    MOI.eval_objective_gradient(model::Manopt.Optimizer, grad, x)
+
+Store the Riemannian gradient at the vector of values `x` for the vectorization
+of the problem.
+"""
+function MOI.eval_objective_gradient(model::Optimizer, grad, x)
```

Comment:
I think I roughly see how this function works – at least the Manopt part of it – again doc string to explain the interplay might be nice.

Reply:
This function has been removed